embed/etcd.go

```diff
@@ -83,6 +83,10 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 		return
 	}
 	if e.sctxs, err = startClientListeners(cfg); err != nil {
+		// errored before starting gRPC server for serveCtx.grpcServerC
+		for _, sctx := range e.sctxs {
+			close(sctx.grpcServerC)
+		}
```

Comment:
maybe a helper function to close sctxs? I see that this code is called twice already.